### PR TITLE
Move Android/Java library code (setuppayload)

### PR DESCRIPTION
**Pull request by joonhaengHeo**
#26815 

Move Android/Java library code

src/controller/onboardingpayload -> src/setup_payload/kotlin

### CODE REVIEW

File: src/tlv/kotlin/BUILD.gn
Comment (bzbarsky-apple) [inline src/tlv/kotlin/BUILD.gn:1]: It's pretty confusing to have `src/tlv` which is not where the actual "normal" tlv code lives.

Is this expected to be used somewhere other than Android?  If not, should it be under src/android or src/platform/android, perhaps?

**Comment by yufengwangca** [comment]
Currently, we have all Java controller APIs under src/controller/java and all java platform APIs under src/platform/android/java

Java controller APIs contains following packages
yufengw@yufengw:~/connectedhomeip/src/controller/java/src/chip$ ls
clusterinfo  devicecontroller  jsontlv  onboardingpayload  tlv

I think it will make the Java API even more confusing if we just move two packages out of src/controller/java, but not others
src/controller/tlv -> src/tlv/kotlin
src/controller/onboardingpayload -> src/setup_payload/kotlin

**Comment by joonhaengHeo** [comment]
> Currently, we have all Java controller APIs under src/controller/java and all java platform APIs under src/platform/android/java
> 
> Java controller APIs contains following packages yufengw@yufengw:~/connectedhomeip/src/controller/java/src/chip$ ls clusterinfo devicecontroller jsontlv onboardingpayload tlv
> 
> I think it will make the Java API even more confusing if we just move two packages out of src/controller/java, but not others src/controller/tlv -> src/tlv/kotlin src/controller/onboardingpayload -> src/setup_payload/kotlin

@yufengwangca 
Yes, I understand.
In my opinion, moving the tlv and jsontlv libraries will require more thought, but I think it's more appropriate to put onboardingpayload under setup_payload. I think it is more appropriate to move the library, which is currently implemented in java, to this side under src/setup_payload.

File: src/tlv/kotlin/BUILD.gn
Comment (joonhaengHeo) [reply src/tlv/kotlin/BUILD.gn:1]: Tlv library location is rollback.

**Comment by andy31415** [comment]
@joonhaengHeo please update the summary to describe *why* we are moving things and how that fixes bits.

Just referencing an open bug is insufficient. I need to understand why the error occurs, why the movement fixes the error.

**Comment by andy31415** [comment]
I have an alternate fix in #26893 that does not move things around.

**Review by andy31415 — CHANGES_REQUESTED** [review]
Moving would have solved it because your kotlin_library now gets generated into a different path, so no more conflicts. However I think the fix in java/rules.gni is more correct.

**Comment by yufengwangca** [comment]
> > Currently, we have all Java controller APIs under src/controller/java and all java platform APIs under src/platform/android/java
> > Java controller APIs contains following packages yufengw@yufengw:~/connectedhomeip/src/controller/java/src/chip$ ls clusterinfo devicecontroller jsontlv onboardingpayload tlv
> > I think it will make the Java API even more confusing if we just move two packages out of src/controller/java, but not others src/controller/tlv -> src/tlv/kotlin src/controller/onboardingpayload -> src/setup_payload/kotlin
> 
> @yufengwangca Yes, I understand. In my opinion, moving the tlv and jsontlv libraries will require more thought, but I think it's more appropriate to put onboardingpayload under setup_payload. I think it is more appropriate to move the library, which is currently implemented in java, to this side under src/setup_payload.

"Setup Payload" is the legacy pre-spec SDK name, it has been renamed to "onboarding payload" in spec. The plan is to rewrite setup_payload fully in kotlin under src/controller/java/src/chip/onboardingpayload and deprecate current src/setup_payload written in java/c++.

**Comment by andy31415** [comment]
I still believe moving/separating logic may make sense (e.g. tlv seems generic and not controller related), however we should not do it for the sake of fixing compile bugs.

**Pull request closed without merging**